controlpanel/api/cluster.py

```diff
@@ -218,230 +114,72 @@ def url(self):
-        }
-        self.role_name = role_name
-
-    def save(self):
-        try:
-            aws.create_role(self.role_name, self.policy)
-        except Exception as e:
-            if not is_ignored_exception(e):
-                raise e
-
-    def allow_assume_role_with_saml(self):
-        self.policy["Statement"].append(dict(SAML_STATEMENT))
-
-    def allow_assume_role_with_oidc(self, sub):
-        stmt = dict(OIDC_STATEMENT)
-        stmt["Condition"] = {
-            "StringEquals": {
-                f"{settings.OIDC_DOMAIN}/:sub": sub,
-            }
-        }
-        self.policy["Statement"].append(stmt)
-
+class S3Bucket:
+    """Wraps a S3Bucket model to provide convenience methods for AWS"""
+    def __init__(self, bucket):
+        self.bucket = bucket
 
-class S3AccessPolicy(dict):
-    POLICY_NAME = "s3-access"
-
-    def __init__(self, role_name=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        self.role_name = role_name
-        self._ro = set()
-        self._rw = set()
-
-        if "Version" not in self:
-            self["Version"] = "2012-10-17",
-
-        for statement in self.get("Statement", []):
-            sid = statement.get("Sid")
+    @property
+    def arn(self):
+        return s3_arn(self.bucket.name)
 
-            if sid == "readonly":
-                self._ro = set([
-                    arn.rstrip('/*')
-                    for arn in statement["Resource"]
-                ])
+    def create(self):
+        return aws.create_bucket(self.bucket.name, self.bucket.is_data_warehouse)
 
-            elif sid == "readwrite":
-                self._rw = set([
-                    arn.rstrip('/*')
-                    for arn in statement["Resource"]
-                ])
 
-    @classmethod
-    def load(cls, role_name):
-        document = aws.get_inline_policy_document(
-            role_name=role_name,
-            policy_name=cls.POLICY_NAME,
-        )
-
-        if document is None:
-            return cls(role_name=role_name)
+class RoleGroup:
+    """
+    Uses a managed policy as a way to group IAM roles that have access to same
+    resources.
 
-        policy = cls(role_name=role_name, **document)
-        return policy
+    This is because IAM doesn't allow adding roles to IAM groups
+    See https://stackoverflow.com/a/48087433/455642
+    """
+    def __init__(self, iam_managed_policy):
+        self.policy = iam_managed_policy
 
-    def __enter__(self):
-        return self
+    @property
+    def arn(self):
+        return iam_arn(f"policy{self.path}{self.policy.name}")
 
-    def __exit__(self, type, value, traceback):
-        self.save()
-        return True
+    @property
+    def path(self):
+        return f'/{settings.ENV}/group/'
 
-    def save(self):
-        aws.put_role_policy(
-            self.role_name,
-            policy_name=self.POLICY_NAME,
-            policy_document=self,
+    def create(self):
+        aws.create_group_policy(
```

Comment:
`create_group_policy` does not exist in `aws`.

Reply:
Yep, I think @xoen already flagged this